pennylane_lightning/lightning_tensor/_tensornet.py

```
@@ -34,53 +34,132 @@
 from pennylane.wires import Wires
 
 
-def svd_split(Mat, site_shape, max_bond_dim):
-    """SVD decomposition of a matrix via numpy linalg. Note that this function is to be moved to the C++ layer."""
+def svd_split(
+    Mat: np.ndarray, site_shape: list, max_bond_dim: int, direction: str
+) -> tuple[np.ndarray, np.ndarray]:
+    """SVD decomposition of a matrix via numpy linalg. Note that this function is to be moved to the C++ layer.
+
+    Args:
+        Mat (array): input matrix
+        site_shape (list): shape of the site tensor
+        max_bond_dim (int): maximum bond dimension
+        direction (str): direction of the SVD decomposition. Options: ["right", "left"]
```

should we throw an error if direction is neither "right" nor "left"

---

can you also clarify what left/right here means? is this what you mean/is the convention: `left` means we keep the left matrix orthnormal, that's why we multiply S to the rhs matrix (Vd)?

---

Alternatively, if it is sensible with the rest of the code style, you can use a boolean `right` or `left` instead of a str direction (whichever is more sensible as a default). This way we only have the 2 expected cases.

---

I mean something like `is_right`, and if it is not, it'll be left.

---

> should we throw an error if direction is neither "right" nor "left"

No, it should be one of them. 

> can you also clarify what left/right here means? 

Ok I will add more information 

> I mean something like `is_right`, and if it is not, it'll be left.

Nice idea, I will included

---

Done!